crates/symbolicator/src/services/symbolication.rs

```diff
@@ -1113,22 +1113,104 @@ fn symbolicate_frame(
     Ok(rv)
 }
 
+/// Stacktrace related Metrics
+///
+/// This gives some metrics about the quality of the stack traces included
+/// in a symbolication request. See the individual members for more information.
+#[derive(Default)]
+struct StacktraceMetrics {
+    /// A truncated stack trace is one that does not end in a
+    /// well known thread base.
+    truncated_traces: u64,
+
+    /// This indicated a stack trace that has at least one bad frame
+    /// from the below categories.
+    bad_traces: u64,
+
+    /// Frames that were scanned.
+    ///
+    /// These are frequently wrong and lead to bad and incomplete stack traces.
+    /// We can improve (lower) these numbers by having more usable CFI info.
+    scanned_frames: u64,
+
+    /// Unsymbolicated Frames.
+    ///
+    /// These may be the result of unavailable or broken debug info.
+    /// We can improve (lower) these numbers by having more usable debug info.
+    unsymbolicated_frames: u64,
+
+    /// Frames referencing unmapped memory regions.
+    ///
+    /// These may be the result of issues in the client-side module finder, or
+    /// broken debug-id information.
+    ///
+    /// We can improve this by fixing client-side implementations and having
+    /// proper debug-ids.
+    unmapped_frames: u64,
+}
+
+impl StacktraceMetrics {
+    fn update(&mut self, other: StacktraceMetrics) {
```

Comment:
This could work well as a `PlusAssign` impl.

Reply:
I thought about it, but IMO this conveys the intent better.

Reply:
Yeah, that's fair.